runtime/compiler/runtime/Listener.cpp

```diff
@@ -45,10 +45,34 @@
 #include "runtime/Listener.hpp"
 
 static bool
-handleOpenSSLConnectionError(int connfd, SSL *&ssl, BIO *&bio, const char *errMsg)
+handleOpenSSLConnectionError(int connfd, SSL *&ssl, BIO *&bio, const char *errMsg, int ret, TR::CompilationInfo *compInfo)
    {
+   int errnoCopy = errno;
+   int sslError = SSL_ERROR_NONE;
+   unsigned long errorCode = 0;
+   int errorReason = 0;
+   char errorString[256] = {0};
+
+   // Filter out SSL_R_UNEXPECTED_EOF_WHILE_READING errors that were introduced in SSL version 3
+   // These can appear if Kubernetes readiness/liveness probes are used on the TCP encrypted channel
+   if (ret <= 0)
+      {
+      sslError = (*OSSL_get_error)(ssl, ret);
+      // Get earliest error code from the thread's error queue without modifying it.
+      errorCode = (*OERR_peek_error)();
+      errorReason = ERR_GET_REASON(errorCode);
+      (*OERR_error_string_n)(errorCode, errorString, sizeof(errorString));
+      if (sslError == SSL_ERROR_SSL && errorReason == SSL_R_UNEXPECTED_EOF_WHILE_READING)
+         {
+         // Take this error out of the queue so that it doesn't get printed
+         errorCode = (*OERR_get_error)();
+         }
+      }
    if (TR::Options::getVerboseOption(TR_VerboseJITServer))
-       TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "%s: errno=%d", errMsg, errno);
+       TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "t=%lu %s: errno=%d sslError=%d errorString=%s",
+                                      (unsigned long)compInfo->getPersistentInfo()->getElapsedTime(),
+                                      errMsg, errnoCopy, sslError, errorString);
+   // Print the error strings for all errors that OpenSSL has recorded and empty the error queue
    (*OERR_print_errors_fp)(stderr);
```

Comment:
Won't this always print the errors in `stderr`? Unless you only care about what's printed on `stdout`.

Reply:
Yes, but you seem to be calling `ERR_peek_error()` a few lines above, which doesn't remove it from the queue. Are you calling the `ERR_get_error()` at some other point that I am missing?

Reply:
The intended logic is to print all errors, except `SSL_R_UNEXPECTED_EOF_WHILE_READING`.
I use `ERR_peek_error()` to determine the error reason, and if that is `SSL_R_UNEXPECTED_EOF_WHILE_READING`, then I use `ERR_get_error()` the eliminate from the queue and prevent it from being printed.

Reply:
Ok, cool. I didn't see the `ERR_get_error()` in the if statement. LGTM otherwise.